macrosynergy/management/utils/sparse.py

```diff
@@ -762,6 +763,62 @@ def to_qdf(
             metrics=metrics,
         )
 
+    def to_dict(
+            self, ticker: str
+        ) -> Dict[str, Union[List[Tuple[str, float, str, float]], Tuple[str, str, str], str]]:
+        # TODO store as arrays instead: [{"data": {"real_date": [], "value": [], "eop": [], "grading": []}}]
+        data = [
+            (f"{index:%Y-%m-%d}", row.value, f"{row.eop:%Y-%m-%d}", row.grading)
+            for index, row in self[ticker][["value", "eop", "grading"]].iterrows()
+        ]
+
+        columns = ("real_date", "value", "eop", "grading")
+        return_dict = {
+            "data": data,
+            "columns": columns,
+            "last_real_date": f"{self._max_period:%Y-%m-%d}",
+            "ticker": ticker
+        }
+        return return_dict
+
+    def to_json(self, ticker: str) -> str:
+        return json.dumps(self.to_dict(ticker))
+
+    def get_releases(
+        self,
+        from_date: pd.Timestamp = pd.Timestamp.today().normalize() - pd.offsets.BDay(1),
+        to_date: pd.Timestamp = pd.Timestamp.today().normalize(),
+        excl_xcats: List[str] = None,
+    ) -> pd.DataFrame:
+        store = []
+        for k, v in self.items():
+            real_date = v.index.max()
+            s = v[v.index == real_date]
```

Comment:
Should this be changed to iterate over each unique release in [from_date, to_date] instead of just the last one?